grafana/files/grafana.ini

```diff
@@ -193,7 +193,7 @@ org_name = {{ server.auth.role }}
 
 #################################### Auth Proxy ##########################
 [auth.proxy]
-{%- if server.auth.engine == 'proxy' %}
+{%- if server.auth.engine == 'proxy' or server.auth.get('proxy', {}).get('enabled', False) %}
```

Comment:
L198 and 199 need to be rewritten to support the legacy auth model and the new one.